hal/armv7m/string.c

```diff
@@ -240,7 +240,7 @@ char *hal_strchr(const char *s, int c)
 }
 
 
-int hal_i2s(char *prefix, char *s, unsigned int i, unsigned char b, char zero)
+__attribute__((section(".noxip"))) int hal_i2s(char *prefix, char *s, unsigned int i, unsigned char b, char zero)
```

Comment:
side discussion: maybe we should switch to 
```c
__attribute__((section(".noxip"))) 
int hal_i2s(char *prefix, char *s, unsigned int i, unsigned char b, char zero)
{
    // ...
}
```
style?

Reply:
The approach you proposed is more readable while browsing a single file, but when searching for definitions and checking if they are `.noxip` it introduces much more friction. However, I don't think that after this PR looking for `.noxip` usages is going to be a common problem.

Reply:
I don't think it's harder (`rg -A1 noxip`), but it depends on the tool i suppose ;)

Reply:
... buuut the `clang-format` cannot be currently configured to put a newline after the attribute - I see that there is a stale PR open which adds the functionality (https://github.com/llvm/llvm-project/pull/78102), so it will take some time to be backed up into stable release of clang-format :)